components/mmu_server.py

```diff
@@ -77,11 +78,31 @@ def __init__(self, config: ConfigHelper):
             self.server.register_remote_method("spoolman_get_spool_info", self.display_spool_info)
             self.server.register_remote_method("spoolman_display_spool_location", self.display_spool_location)
 
-        # Replace file_manager/metadata with this file
-        self.setup_placeholder_processor(config)
-
         # Options
         self.update_location = self.config.getboolean("update_spoolman_location", True)
+        self.proc_config: Dict[str, Any] = {
+            "name": "mmu_server",
+            "command": " ".join([
+                sys.executable,
+                os.path.abspath(__file__),
+                "-f",
+                "{gcode_file_path}",
+                "-m" if config.getboolean("enable_file_preprocessor", True) else "",
+                "-n" if config.getboolean("enable_toolchange_next_pos", True) else "",
+            ]),
+            "timeout": 300,
+            "version": config.getfloat('happy_hare_version', 2.2),
+            "ident": {
```

Comment:
The "ident" values are used by metadata.py as a way of managing state so the processor only runs once.